pkg/api/customization/multiclusterapp/multi_cluster_app.go

```diff
@@ -93,6 +93,7 @@ func (w Wrapper) addProjects(request *types.APIContext) error {
 			existingProjects[p] = true
 		}
 		mcappToUpdate = mcapp.DeepCopy()
+		mcappToUpdate.Spec.TargetToRole = targetsToRole
```

Comment:
What if mcappToUpdate.Spec.TargetToRole is already non-empty - do we need to recalculate the mcappToUpdate.Spec.TargetToRole for all targets on the mcapp and not just the new projects?

Reply:
We won't need to check roles in existing targets, because while adding projects we just want to get caller's roles in their new projects. 
mcappToUpdate.Spec.TargetToRole should be overridden so now it only holds targets that are being newly added. Because the roles for existing project were already granted to the system account. Does this sound okay?

Reply:
@mrajashree is there a rancher/types PR to add this field: mcappToUpdate.Spec.TargetToRole

Reply:
@prachidamle yes, sorry I forgot to include it, here it is https://github.com/rancher/types/pull/745